### PR TITLE
Added "get" and "get_mut" to BaseMatrix and BaseMatrixMut

### LINKED ISSUE #151: BaseMatrix has no `get` method.
As the title says, `BaseMatrix` has no `get` method, only `get_unchecked`. Is this deliberate or just an omission? I'm rewriting some generic code (mainly forward and back substitution) and I wanted to avoid unsafe code.

### PR DESCRIPTION
**Pull request by prismatica**
This resolves #151, unless you all also want to require an Index bound for BaseMatrix (which I didn't add).

### CODE REVIEW

File: src/matrix/base/mod.rs [hunk 1]
```diff
@@ -80,6 +80,35 @@ pub trait BaseMatrix<T>: Sized {
...
+    ///
+    /// assert_eq!( *mat.get([0, 0]).unwrap(), 0)
+    /// # }
+    /// ```
+    fn get(&self, index: [usize; 2]) -> Option<&T> {
+				let row_ind = index[0];
+				let col_ind = index[1];
+				
+        if row_ind >= self.rows() || col_ind >= self.cols() {
+					None
```
Comment (AtheMathmo) [inline src/matrix/base/mod.rs]: Formatting here looks a little off - indentation is too large?

**Comment by tafia** [comment]
No big deal and not directly related to this PR, but I was wondering if anyone tried adding some `#[inline]` attributes in these kind of functions?

File: src/matrix/base/mod.rs [hunk 2]
```diff
@@ -80,6 +80,35 @@ pub trait BaseMatrix<T>: Sized {
         &*(self.as_ptr().offset((index[0] * self.row_stride() + index[1]) as isize))
     }
 
+    /// Get a reference to a point in the matrix.
```
Comment (Andlon) [inline src/matrix/base/mod.rs]: I feel that "point" is a little strange. However, I see that this is what the existing docs for `get_unchecked` say too, so you've certainly done the right thing by using the same language.

@AtheMathmo: Perhaps this should be changed to "an element" for both `get` and `get_unchecked` (and their `mut` brethren)?

**Comment by Andlon** [comment]
@tafia: My understanding is somewhat limited in this regard, but I would expect it not to have any particular effect on generic functions, since they must be effectively "imported" into the compilation unit of the user anyway.

That is, my understanding of `#[inline]` is primarily that it tells the compiler that external crates may want to inline this function. However, for generic functions, I would expect the whole function to get monomorphized at the call site instead, in which case the compiler is able to inline it if it finds it appropriate.

Do take it with a large pinch of salt, however. I might be completely wrong here :)

File: src/matrix/base/mod.rs (hunk 1 above)
Comment (prismatica) [inline src/matrix/base/mod.rs]: Yup, definitely! I can fix that tonight.

File: src/matrix/base/mod.rs (hunk 2 above)
Comment (prismatica) [inline src/matrix/base/mod.rs]: I also think that "point" is unusual, for what it's worth. I'd be happy to update the comments, if that's the consensus.

Comment (AtheMathmo) [inline src/matrix/base/mod.rs]: Yeah let's change all of these to `element`. I'm not sure what I was thinking by `point` :)

**Comment by AtheMathmo** [comment]
I'm also not sure if there is any benefit to inlining these functions. I've never compared inlining with generic functions but what @Andlon says makes sense.

Perhaps this is worth investigating separately to this PR though.

**Comment by prismatica** [comment]
Are there any changes I should make to this latest merge?

**Comment by AtheMathmo** [comment]
At a glance it looks good to me. I'll try to give it a proper look later tonight (I'm just about to head out).

Sorry for the delay - it's been a very busy week for me.

**Comment by prismatica** [comment]
No problem at all, didn't mean to rush you!

**Pull request merged**